File: Linked List/SLL.c
//Linked list implementation in C
#include <stdio.h>
#include <stdlib.h>
    
struct Node{
    int data;
    struct Node * next;
};
typedef struct Node SLLNODE;

SLLNODE* create_SLLNode(int input){
    SLLNODE* newNode;
    newNode = (SLLNODE*)malloc(sizeof(SLLNODE));
    newNode->data=input;
    newNode->next=NULL;
    return newNode;
}

void insertBegin(SLLNODE* *start,int input){
    SLLNODE* newNode = create_SLLNode(input);
    if(start == NULL) *start = newNode;
    else{
        newNode->next = *start;
        *start = newNode;
    }
}

void insertEnd(SLLNODE* *start,int input){
    SLLNODE* newNode = create_SLLNode(input);
    if(*start == NULL) *start = newNode;
    else{
		 SLLNODE* ptr;
		 for(ptr = *start; ptr->next !=NULL; ptr=ptr->next);              ptr->next = newNode;
	 }
}

void deleteBegin(SLLNODE* *start){
    if(*start == NULL) printf("The linked list empty");
	 else{
		 SLLNODE* toFree;
	    toFree = *start;
		 *start = (*start)->next;
		 free(toFree);
	 }
}

void deleteEnd(SLLNODE* *start)
{
    if(*start == NULL) printf("The linked list empty");
	 else{
	   SLLNODE* ptr;
		SLLNODE* prev;
		for(ptr=*start; ptr->next!=NULL; ptr=ptr->next){
		  prev = ptr;
		}
		prev->next = NULL;
		free(ptr);
	 }
}

void displaySLL(SLLNODE* start)
{
    SLLNODE* ptr; 
    for(ptr = start; ptr!=NULL; ptr=ptr->next){
        printf("%d ",ptr->data);    
    }
    printf("\n");
}

int main(){
    int input;
    SLLNODE* start = NULL;
    printf("Enter numbers to create a SLL: \n");
    while(1){
        scanf("%d",&input);
        if(input == -1) break;
        insertEnd(&start,input);
    }
	 deleteEnd(&start);
    displaySLL(start);
}

File: Linked List/SLL.py
class Node:
	def __init__(self,data = None):
		self.data = data
		self.next = None
	
class LinkedList:
	'''
		Linked List
		-----------------
		Methods available: 
		display()
		insertBegin(num)
		insertEnd(num)
		insertAt(pos,num)
		deleteBegin()
		deleteEnd()
		deleteAt(pos,num)
    '''
	#default constructor method to create a start(head)
	def __init__(self):
		self.start = None
	
	#method to print the data in Linked List
	def display(self):
		ptr = self.start
		while (ptr != None):
			print(ptr.data,end=" ")
			ptr=ptr.next
		print("\n",end="")
			
	#method to insert data at beginning in Linked List		
	def insertBegin(self,input):
		newNode = Node(input)
		if(self.start == None):
			self.start = newNode
		else:
			newNode.next  = self.start
			self.start = newNode
			
	#method to insert data at end in Linked List
	def insertEnd(self,input):
		newNode = Node(input)
		if(self.start == None):
			self.start = newNode
		else:
			ptr = self.start
			while(ptr.next!=None):
				ptr = ptr.next
			ptr.next = newNode
			
	#method to insert data at a position in Linked List
	def insertAt(self,pos,input):
		pass
	
	
	#method to delete data at beginning in Linked List
	def deleteBegin(self):
		if(self.start == None):
			print("Linked list is empty")
		else:
			temp = self.start
			self.start = self.start.next
			temp = None
	
	#method to delete data at end in Linked List
	def deleteEnd(self):
		if(self.start == None):
			print("Linked list is empty")
		else:
			ptr = self.start
			prev = None
			while(ptr.next != None):
				prev = ptr
				ptr = ptr.next
			prev.next = None
			ptr = None
	
	#method to delete data at a position in Linked List
	def deleteAt(self,pos,input):
		pass

#Main program			
if __name__ == "__main__":
	LL = LinkedList() #LL object created for LinkedList class
	num = list(map(int, input().split()))
	for it in range(len(num)):
		LL.insertEnd(num[it])
	LL.deleteEnd()
	LL.display()
	#print(LL.__doc__) #to print the docstring
	